### PR TITLE
fix: [CDS-48979]: add more log to give more context when it occur again

### PR DESCRIPTION
**Pull request by rafaelmart1ns**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)

### CODE REVIEW

**Review by vaibhav-si — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java` line 3424 at r1](https://reviewable.io/reviews/harness/harness-core/41933#-NJxa3gJ41YHJm_yzqmt:-NJxa3gJ41YHJm_yzqmu:budzzk) ([raw file](https://github.com/harness/harness-core/blob/84a3d20c49d41d5af64a2593fe0162e076009a58/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L3424)):*
> ```Java
>         requiredEntityTypes.add(HELM_CHART);
>       }
>     } catch (NullPointerException e) {
> ```

I think better way to do this by checking for null/empty serviced and throwing this message then.

NullPointer can happen due to anything in above code block. It's not even future proof if more lines are added in try block.

**Review by Fernando-Dourado — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java` line 3429 at r1](https://reviewable.io/reviews/harness/harness-core/41933#-NJyPSrb9hpxoN1I__eR:-NJyPSrb9hpxoN1I__eS:bsrn3le) ([raw file](https://github.com/harness/harness-core/blob/84a3d20c49d41d5af64a2593fe0162e076009a58/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L3429)):*
> ```Java
>           serviceId, workflowPhase);
>       log.error(error);
>       throw new UndefinedValueException("This serviceId no longer supported: " + serviceId);
> ```

This error message is not clear. The serviced is missing, and not "no longer supported".

**Review by rafaelmart1ns — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java` line 3424 at r1](https://reviewable.io/reviews/harness/harness-core/41933#-NJxa3gJ41YHJm_yzqmt:-NJysNvB1dmVQq7H7rew:bo8st0y) ([raw file](https://github.com/harness/harness-core/blob/84a3d20c49d41d5af64a2593fe0162e076009a58/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L3424)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

I think better way to do this by checking for null/empty serviced and throwing this message then.

NullPointer can happen due to anything in above code block. It's not even future proof if more lines are added in try block.
</blockquote></details>

Make sense. I updated it. Check again.
___
*[`400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java` line 3429 at r1](https://reviewable.io/reviews/harness/harness-core/41933#-NJyPSrb9hpxoN1I__eR:-NJysR8J9wMxry-qh0T-:b-6soc7e) ([raw file](https://github.com/harness/harness-core/blob/84a3d20c49d41d5af64a2593fe0162e076009a58/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L3429)):*
<details><summary><i>Previously, Fernando-Dourado (Fernando Dourado) wrote…</i></summary><blockquote>

This error message is not clear. The serviced is missing, and not "no longer supported".
</blockquote></details>

it changed.

**Review by rafaelmart1ns — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

File: 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java [hunk 1]
```diff
@@ -3290,6 +3291,10 @@ private boolean requiresManifest(String serviceId, String appId, List<String> po
       return false;
     }
     Service service = serviceResourceService.get(appId, serviceId);
+    if (service == null) {
+      log.error("ServiceResourceService is empty or invalid for appId {}, serviceId {}.", appId, serviceId);
+      throw new UndefinedValueException("This serviceId is empty or invalid: " + serviceId);
```
Comment (lasdox) [inline 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:3296]:
```suggestion
      log.error("ServiceId: {} is empty or invalid for appId {}.", serviceId, appId);
      throw new UndefinedValueException("ServiceId is empty or invalid: {} - Please check your workflow settings", serviceId);
```

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review.

**Review by Fernando-Dourado — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java` line 3295 at r3](https://reviewable.io/reviews/harness/harness-core/41933#-NJzkzA74tok2_AhKq6Z:-NJzkzA74tok2_AhKq6_:b7zlfg6) ([raw file](https://github.com/harness/harness-core/blob/c46672596086f3cc46368600c9e753f690377edf/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L3295)):*
> ```Java
>     Service service = serviceResourceService.get(appId, serviceId);
>     if (service == null) {
>       String messageError = String.format("ServiceId: {} is empty or invalid for appId {}.", serviceId, appId);
> ```

The message format is wrong, instead of {} you need to use %s to replace it with a String value.

The {} is a replace pattern valid only for logging operations.

**Review by Fernando-Dourado — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by Fernando-Dourado — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by lasdox — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41933)* status: :shipit: complete! ✓

**Pull request merged**